safemysql.class.php

```diff
@@ -608,6 +612,12 @@ private function retryIfDeadlocked($callback)
 				} else {
 					throw $e;
 				}
+			} catch (Throwable $e) {
```

Comment:
- [ ] Wait, no - now that you've restored the `catch` blocks in `transaction()`, this block isn't needed. You've doubly-fixed the bug.